encoding/protobuf.go

```diff
@@ -106,6 +108,50 @@ func (p protoSerializer) Encoding() Encoding {
 	return Protobuf
 }
 
+func (p protoSerializer) ErrorDetails(err error) ([]interface{}, error) {
+	// Here we use yarpcerrors since the transport layer of yab is using yarpc as well
+	if err == nil || !yarpcerrors.IsStatus(err) {
+		return nil, nil
+	}
+
+	details := []interface{}{}
+	yerr := yarpcerrors.FromError(err)
+	if len(yerr.Details()) == 0 {
+		return nil, nil
+	}
+
+	errStatus := &status.Status{}
+	if err := proto.Unmarshal(yerr.Details(), errStatus); err != nil {
+		return nil, fmt.Errorf("could not unmarshal error details %s", err.Error())
```

Comment:
do you want to wrap error with %w?
`fmt.Errorf("could not unmarshal error details %w", err)`
https://blog.golang.org/go1.13-errors

Reply:
we never used %w elsewhere in this project. I set %s to stay consistent with the other errors.

Reply:
ok I can buy this